### PR TITLE
Added Py solution for livestock lineup

### PR DESCRIPTION
**Pull request by AAkashyap1**

### CODE REVIEW

File: solutions/usaco-965.mdx [hunk 1]
```diff
@@ -2,11 +2,14 @@
 id: usaco-965
 source: USACO Bronze 2019 December
 title: Livestock Lineup
-author: Benjamin Qi
+authors: Benjamin Qi, Ananth Kashyap
```
Comment (maggie-j-liu) [inline solutions/usaco-965.mdx:5]:
```suggestion
author: Benjamin Qi, Ananth Kashyap
```
i think `author` is the correct field

File: solutions/usaco-965.mdx [hunk 2]
```diff
@@ -88,4 +91,51 @@ int main() {
...
+
+<PySection>
+## Brute Force with Itertools
+
+<Warning>
+Due to the brute force nature of this solution, it is very slow and is not the optimal solution.
+</Warning>
+We can easily iterate through every possible permutation of cows in lexographic order using Itertools. Then we can check for the first permutation that meets the adjacency constraints and this will be our answer. 
+
+**Time Complexity:** $\mathcal{O}(NC*C!)$, where $C$ represents the number of cows
```
Comment (maggie-j-liu) [inline solutions/usaco-965.mdx:105]:
```suggestion
**Time Complexity:** $\mathcal{O}(NC \cdot C!)$, where $C$ represents the number of cows 
```
also, shouldn't this just be N * C! ?

File: solutions/usaco-965.mdx [hunk 3]
```diff
@@ -88,4 +91,51 @@ int main() {
...
+<PySection>
+## Brute Force with Itertools
+
+<Warning>
+Due to the brute force nature of this solution, it is very slow and is not the optimal solution.
+</Warning>
+We can easily iterate through every possible permutation of cows in lexographic order using Itertools. Then we can check for the first permutation that meets the adjacency constraints and this will be our answer. 
+
+**Time Complexity:** $\mathcal{O}(NC*C!)$, where $C$ represents the number of cows 
+```py
```
Comment (maggie-j-liu) [inline solutions/usaco-965.mdx:106]: use tabs instead of spaces to indent

File: solutions/usaco-965.mdx [hunk 4]
```diff
@@ -88,4 +91,51 @@ int main() {
...
+NUM_COWS = 8
+from itertools import permutations
+
```
Comment (maggie-j-liu) [inline solutions/usaco-965.mdx:109]: i'm not sure about python conventions, but shouldn't import statements usually come first?

**Review by maggie-j-liu — CHANGES_REQUESTED** [review]

File: solutions/usaco-965.mdx (hunk 2 above)
Comment (AAkashyap1) [reply solutions/usaco-965.mdx:105]: I use .index tho so isnt it multiplied by C?

File: solutions/usaco-965.mdx (hunk 4 above)
Comment (AAkashyap1) [reply solutions/usaco-965.mdx:109]: uhh honestly i have no idea @SansPapyrus683 @ryanchou-dev

Comment (SansPapyrus683) [reply solutions/usaco-965.mdx:109]: yeah imports come first

Comment (AAkashyap1) [reply solutions/usaco-965.mdx]: k

File: solutions/usaco-965.mdx [hunk 5]
```diff
@@ -88,4 +91,51 @@ int main() {
...
+<Warning>
+Due to the brute force nature of this solution, it is very slow and is not the optimal solution.
+</Warning>
+We can easily iterate through every possible permutation of cows in lexographic order using Itertools. Then we can check for the first permutation that meets the adjacency constraints and this will be our answer. 
+
+**Time Complexity:** $\mathcal{O}(NC \cdot C!)$, where $C$ represents the number of cows 
+```py
+from itertools import permutations
+NUM_COWS = 8
```
Comment (SansPapyrus683) [inline solutions/usaco-965.mdx:108]:
```suggestion

NUM_COWS = 8
```

File: solutions/usaco-965.mdx [hunk 6]
```diff
@@ -88,4 +91,51 @@ int main() {
...
+
+data = open('lineup.in').read().strip().split('\n')
+n = int(data[0])
+
+cows = sorted(['Bessie', 'Buttercup', 'Belinda', 'Beatrice', 
+			      'Bella', 'Blue', 'Betsy', 'Sue'])
```
Comment (SansPapyrus683) [inline solutions/usaco-965.mdx:114]: the indentation is kinda weird here

File: solutions/usaco-965.mdx [hunk 7]
```diff
@@ -88,4 +91,51 @@ int main() {
...
+adjacents = []
+
+for i in range(n):
+    temp = data[i + 1].split(' ')
+    adjacents.append([temp[0], temp[-1]])
+
+ans = ''
+
```
Comment (SansPapyrus683) [inline solutions/usaco-965.mdx:122]:
```suggestion
```

File: solutions/usaco-965.mdx [hunk 8]
```diff
@@ -88,4 +91,51 @@ int main() {
...
+orderings = [list(permutation) for permutation in list(permutations(cows))]
+
```
Comment (SansPapyrus683) [inline solutions/usaco-965.mdx:124]:
```suggestion
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-965.mdx (hunk 6 above)
Comment (AAkashyap1) [reply solutions/usaco-965.mdx:115]: tell me how to change master

File: solutions/usaco-965.mdx (hunk 3 above)
Comment (maggie-j-liu) [reply solutions/usaco-965.mdx:106]: not resolved?

Comment (AAkashyap1) [reply solutions/usaco-965.mdx:106]: wait i thought i did this

Comment (AAkashyap1) [reply solutions/usaco-965.mdx:106]: let me try again

Comment (AAkashyap1) [reply solutions/usaco-965.mdx:106]: does it look good now @maggie-j-liu ?

File: solutions/usaco-965.mdx [hunk 9]
```diff
@@ -88,4 +91,50 @@ int main() {
...
+**Time Complexity:** $\mathcal{O}(NC \cdot C!)$, where $C$ represents the number of cows 
+```py
+from itertools import permutations
+
+NUM_COWS = 8
+
+data = open('lineup.in').read().strip().split('\n')
+n = int(data[0])
+
+cows = sorted(['Bessie', 'Buttercup', 'Belinda', 'Beatrice',
```
Comment (SansPapyrus683) [inline solutions/usaco-965.mdx:114]: wait
instead of NUM_COWS why not just use len(cows) lol

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
yeah sure whatever

File: solutions/usaco-965.mdx (hunk 9 above)
Comment (AAkashyap1) [reply solutions/usaco-965.mdx:114]: lol originally i just had 8 but i thought abt constants

Comment (SansPapyrus683) [reply solutions/usaco-965.mdx:114]: so you gonna change this or like

File: solutions/usaco-965.mdx [hunk 10]
```diff
@@ -88,4 +91,50 @@ int main() {
...
+ans = ''
+orderings = [list(permutation) for permutation in list(permutations(cows))]
+for i in orderings:
+	count = 0
+	for j in adjacents:
+		if abs(i.index(j[0]) - i.index(j[1])) == 1:
+			count += 1
+	if count == n:
+		for k in range(NUM_COWS):
+			if k != 7:
```
Comment (SansPapyrus683) [inline solutions/usaco-965.mdx:131]: 7 is a magic number here

File: solutions/usaco-965.mdx (hunk 9 above)
Comment (AAkashyap1) [reply solutions/usaco-965.mdx:114]: ok sure

File: solutions/usaco-965.mdx (hunk 10 above)
Comment (AAkashyap1) [reply solutions/usaco-965.mdx:129]: len(cows) - 1 ill replace

**Review by SansPapyrus683 — APPROVED** [review]

**Review by nathangong — APPROVED** [review]

**Review by dongliu0426 — CHANGES_REQUESTED** [review]
it would be nice if you added some comments in your code

File: solutions/usaco-965.mdx [hunk 11]
```diff
@@ -88,4 +91,48 @@ int main() {
 		cout << cows[t] << '\n';
 	}
 }
-```
\ No newline at end of file
+```
+</CPPSection>
+
+<PySection>
+## Brute Force with Itertools
```
Comment (ryanchou-dev) [inline solutions/usaco-965.mdx:98]: wait move out of language section?

**Review by ryanchou-dev — CHANGES_REQUESTED** [review]
nice, just the review below. i brute forced all permutations to :)

File: solutions/usaco-965.mdx [hunk 12]
```diff
@@ -88,4 +91,48 @@ int main() {
...
+cows = sorted(['Bessie', 'Buttercup', 'Belinda', 'Beatrice', 
+		'Bella', 'Blue', 'Betsy', 'Sue'])
+adjacents = []
+
+for i in range(n):
+	temp = data[i + 1].split(' ')
+	adjacents.append([temp[0], temp[-1]])
+
+ans = ''
+orderings = [list(permutation) for permutation in list(permutations(cows))]
```
[the remark this answers is not in the document]
Comment (AAkashyap1) [reply solutions/usaco-965.mdx:121]: so leave a comment here?

File: solutions/usaco-965.mdx (hunk 11 above)
Comment (jessechoe10) [reply solutions/usaco-965.mdx:98]:
```suggestion
```

File: solutions/usaco-965.mdx [hunk 13]
```diff
@@ -88,4 +91,48 @@ int main() {
...
\ No newline at end of file
+```
+</CPPSection>
+
+<PySection>
+## Brute Force with Itertools
+
+<Warning>
+Due to the brute force nature of this solution, it is very slow and is not the optimal solution.
+</Warning>
```
Comment (jessechoe10) [inline solutions/usaco-965.mdx:102]: Do we really need this?

File: solutions/usaco-965.mdx [hunk 14]
```diff
@@ -88,4 +91,48 @@ int main() {
...
+We can easily iterate through every possible permutation of cows in lexographic order using Itertools. Then we can check for the first permutation that meets the adjacency constraints and this will be our answer.
```
Comment (jessechoe10) [inline solutions/usaco-965.mdx:103]:
```suggestion
We can iterate through every possible permutation of cows in lexographic order using `Itertools`. Then we can check for the first permutation that meets the adjacency constraints and this will be our answer. 
```

**Review by jessechoe10 — CHANGES_REQUESTED** [review]

File: solutions/usaco-965.mdx (hunk 13 above)
Comment (ryanchou-dev) [reply solutions/usaco-965.mdx:102]: yeah i think this can be removed

Comment (AAkashyap1) [reply solutions/usaco-965.mdx:101]: wait nooo

Comment (AAkashyap1) [reply solutions/usaco-965.mdx:101]: i think we should keep

Comment (jessechoe10) [reply solutions/usaco-965.mdx:101]: We should remove b/c the previous solution states "What if there were up to $10^5$ cows? Then neither of the solutions provided in the analysis would be fast enough." Imo, this is pretty redundant, so please remove.

File: solutions/usaco-965.mdx (hunk 12 above)
[the remark this answers is not in the document]
Comment (dongliu0426) [reply solutions/usaco-965.mdx:120]: Change it.

[the remark this answers is not in the document]
Comment (dongliu0426) [reply solutions/usaco-965.mdx:120]:
```suggestion
orderings = list(permutations(cows))
```

File: solutions/usaco-965.mdx (hunk 13 above)
Comment (nwatx) [reply solutions/usaco-965.mdx:101]: I think if you change it to an info section and then say something along the lines of "This solution can be done in {actual tc}"

File: solutions/usaco-965.mdx [hunk 15]
```diff
@@ -88,4 +91,47 @@ int main() {
...
+```
+</CPPSection>
+
+<PySection>
+
+<Warning>
+Due to the brute force nature of this solution, it is very slow and is not the optimal solution.
+</Warning>
+We can iterate through every possible permutation of cows in lexographic order using `Itertools`. Then we can check for the first permutation that meets the adjacency constraints and this will be our answer. 
+
```
Comment (ryanchou-dev) [inline solutions/usaco-965.mdx:103]:
```suggestion
We can iterate through every possible permutation of cows in lexographic order using `Itertools`. Then we can check for the first permutation that meets the adjacency constraints. 

```

**Review by ryanchou-dev — CHANGES_REQUESTED** [review]
